parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -497,7 +523,30 @@ pub mod pallet {
 			let beacon_block_root: H256 =
 				merkleization::hash_tree_root_beacon_header(header.clone())
 					.map_err(|_| Error::<T>::HeaderHashTreeRootFailed)?
-					.into(); // TODO check denylist headers
+					.into();
+
+			let finalized_block_root_hash =
+				<FinalizedBeaconHeadersBlockRoot<T>>::get(last_finalized_header.beacon_block_root);
```

Comment:
IMHO `finalized_block_root_hash` a little confusing here, would it better to rename as `block_roots_root_hash` or  some other to repensent not the root of the beacon block but the root of the blockRoots in the beacon state?

Reply:
Also seems we change to always use the `last_finalized_header` for the verification, does that mean the original `FinalizedBeaconHeaders` storage can be safely removed now?

since it is related to the impl of https://linear.app/snowfork/issue/SNO-328 so I may pause that one until we finalize this PR

Reply:
> Also seems we change to always use the last_finalized_header for the verification, does that mean the original FinalizedBeaconHeaders storage can be safely removed now?

I might still change this, because if we need to import older headers we may have to use an older finalized header for the block proof. Working on this at the moment.